pinecone/index_connection.go

```diff
@@ -99,9 +99,9 @@ type ListVectorsRequest struct {
 }
 
 type ListVectorsResponse struct {
-	VectorIds           []*string
-	Usage               *Usage
-	NextPaginationToken *string
+	VectorIds           []*string `json:"vector_ids,omitempty"`
```

Comment:
Re: your question in the PR description, from what I've read at least it seems like adding `omitempty` is a good call. I think it avoids ambiguous values -- like is a vector ID `0`, or does it not exist? I'm always a fan of being explicit

Reply:
I think there are places where not using `omitempty` does make sense as it'll drop zero values that maybe you want to include.

For this PR I was mainly focused on making sure we're using `omitempty` for pointers and structs where it makes sense, primarily to avoid `null` values in the resulting JSON.